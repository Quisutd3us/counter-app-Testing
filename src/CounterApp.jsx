import {useState} from "react";
import PropTypes from 'prop-types';

const CounterApp = ({value}) => {
      const [counter, setCounter] = useState(value)
      /*sum +1 counter*/
      const addCounter = () => setCounter(c => c + 1)
      /*rest -1 counter*/
      const restCounter = () => counter > 0 ? setCounter(counter - 1) : setCounter(0)
      /*assign counter to 0*/
      const resetCounter = () => setCounter(value)

      return (
          <div>
            <h1>Counter APP</h1>
            <h2>{counter}</h2>
            <div className={'buttonContainer'}>
              <button
                  aria-label={'btnAdd'}
                  onClick={addCounter}
              >+1
              </button>
              <button
                  aria-label={'btnRest'}
                  onClick={restCounter}
              >-1
              </button>
              <button
                  aria-label={'btnReset'}
                  onClick={resetCounter}
              >Reset
              </button>

            </div>

          </div>
      );
    }
;

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;