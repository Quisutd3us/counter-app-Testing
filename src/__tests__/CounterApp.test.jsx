import {fireEvent, render, screen} from "@testing-library/react";
import CounterApp from "../CounterApp.jsx";

describe('Set of testing functions to Counter React App', () => {
  const value = 100
  test('Must be match with the snapshot', () => {
    const {container} = render(
        <CounterApp
            value={value}
        />
    )
    expect(container).toMatchSnapshot()
  })
  test('Must be Show the initial value={100}', () => {
    render(
        <CounterApp
            value={value}
        />
    )
    expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain('100')
  })

  test('Must be increment the value in +1', () => {
    render(
        <CounterApp
            value={value}
        />
    )
    fireEvent.click(screen.getByRole('button', {name: 'btnAdd'}))
    expect(screen.getByText(101)).toBeTruthy()
  })
  test('Must be decremet the value in -1', () => {
    render(
        <CounterApp
            value={value}
        />
    )
    fireEvent.click(screen.getByRole('button', {name: 'btnRest'}))
    fireEvent.click(screen.getByRole('button', {name: 'btnRest'}))
    fireEvent.click(screen.getByRole('button', {name: 'btnRest'}))
    expect(screen.getByText(97)).toBeTruthy()
  })
  test('Must be reset the state for the prop value', () => {
    render(
        <CounterApp
            value={value}
        />
    )
    fireEvent.click(screen.getByRole('button', {name: 'btnReset'}))
    expect(screen.getByText(100)).toBeTruthy()
  })
})