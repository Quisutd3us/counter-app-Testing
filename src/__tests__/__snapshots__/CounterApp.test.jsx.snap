// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set of testing functions to Counter React App Must be match with the snapshot 1`] = `
<div>
  <div>
    <h1>
      Counter APP
    </h1>
    <h2>
      100
    </h2>
    <div
      class="buttonContainer"
    >
      <button
        aria-label="btnAdd"
      >
        +1
      </button>
      <button
        aria-label="btnRest"
      >
        -1
      </button>
      <button
        aria-label="btnReset"
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;
